nightlies_watcher/main.py

```diff
@@ -0,0 +1,41 @@
+import asyncio
+import logging
+import taskcluster
+
+from nightlies_watcher.config import get_config
+from nightlies_watcher.worker import start_message_queue_worker
+
+logger = logging.getLogger(__name__)
+
+
+def main(name=None):
+    if name not in ('__main__', None):
+        return
+
+    config = get_config()
+
+    FORMAT = '%(asctime)s - %(filename)s - %(levelname)s - %(message)s'
+    level = logging.DEBUG if config.get('verbose', False) else logging.INFO
+    logging.basicConfig(format=FORMAT, level=level)
+    logging.getLogger('requests').setLevel(logging.WARNING)
+    logging.getLogger('taskcluster').setLevel(logging.WARNING)
+    logging.getLogger('mohawk').setLevel(logging.WARNING)
+
+    taskcluster.config['credentials']['clientId'] = config['credentials']['client_id']
+    taskcluster.config['credentials']['accessToken'] = config['credentials']['access_token']
+
+    event_loop = asyncio.get_event_loop()
+
+    try:
+        event_loop.run_until_complete(start_message_queue_worker(config))
+        event_loop.run_forever()
```

Comment:
I don't believe it works. I deleted the line, and got these logs:
```
2016-10-20 15:11:15,655 - worker.py - INFO - Processing revision "95567c6551ea7dfef517e7b3f50fe0be209a3a2a" from repository "mozilla-aurora" (triggered by completed task "ThLJb4maQI6gINIH7af5yA")
2016-10-20 15:11:21,267 - publish.py - DEBUG - ['public/build/fennec-51.0a2.en-US.android-i386.apk']
2016-10-20 15:11:21,267 - publish.py - DEBUG - ['public/build/fennec-51.0a2.en-US.android-arm.apk']
2016-10-20 15:11:24,666 - publish.py - INFO - Created task OQY7-tfmQPqBu7KuIgznqw: {'status': {'taskId': 'OQY7-tfmQPqBu7KuIgznqw', 'schedulerId': '-', 'taskGroupId': 'OQY7-tfmQPqBu7KuIgznqw', 'retriesLeft': 0, 'workerType': 'pushapk-v1', 'expires': '2017-10-20T14:11:22.208Z', 'provisionerId': 'scriptworker-prov-v1', 'deadline': '2016-10-20T14:11:22.208Z', 'runs': [{'state': 'pending', 'reasonCreated': 'scheduled', 'scheduled': '2016-10-20T13:11:24.234Z', 'runId': 0}], 'state': 'pending'}}
2016-10-20 15:11:24,667 - worker.py - INFO - Marking message as read. Processed task "ThLJb4maQI6gINIH7af5yA"
2016-10-20 15:11:24,690 - base_events.py - ERROR - Task was destroyed but it is pending!
task: <Task pending coro=<AmqpProtocol.run() running at /home/jlorenzo/.virtualenvs/nightlies-watcher/lib/python3.5/site-packages/aioamqp/protocol.py:284> wait_for=<Future pending cb=[Task._wakeup()]>>
2016-10-20 15:11:24,690 - base_events.py - ERROR - Fatal write error on socket transport
protocol: <asyncio.sslproto.SSLProtocol object at 0x7fa0f5471c18>
transport: <_SelectorSocketTransport fd=7>
Traceback (most recent call last):
  File "/usr/lib64/python3.5/asyncio/selector_events.py", line 707, in write
    n = self._sock.send(data)
OSError: [Errno 9] Bad file descriptor
Exception ignored in: <bound method _SSLProtocolTransport.__del__ of <asyncio.sslproto._SSLProtocolTransport object at 0x7fa0f5488d68>>
Traceback (most recent call last):
  File "/usr/lib64/python3.5/asyncio/sslproto.py", line 328, in __del__
  File "/usr/lib64/python3.5/asyncio/sslproto.py", line 319, in close
  File "/usr/lib64/python3.5/asyncio/sslproto.py", line 542, in _start_shutdown
  File "/usr/lib64/python3.5/asyncio/sslproto.py", line 547, in _write_appdata
  File "/usr/lib64/python3.5/asyncio/sslproto.py", line 651, in _process_write_backlog
  File "/usr/lib64/python3.5/asyncio/sslproto.py", line 658, in _fatal_error
NameError: name 'base_events' is not defined
```

There was still one message to handle in the queue, and it hasn't been processed in the run. I had to restart the process. Same thing happened at the end of my second run.

If the queue is empty, the process ends like this:
```
2016-10-20 15:12:24,388 - selector_events.py - DEBUG - Using selector: EpollSelector
2016-10-20 15:12:25,801 - protocol.py - INFO - Recv open ok
2016-10-20 15:12:26,006 - channel.py - DEBUG - Channel is open
2016-10-20 15:12:26,168 - channel.py - DEBUG - Qos ok
2016-10-20 15:12:26,169 - worker.py - INFO - Using queue: queue/nightlies-watcher-test/nightlies-watcher-test
2016-10-20 15:12:26,411 - channel.py - DEBUG - Queue declared
2016-10-20 15:12:26,411 - worker.py - INFO - Binding {'path': 'exchange/taskcluster-queue/v1/task-completed', 'routing_keys': ['route.index.gecko.v2.mozilla-aurora.nightly.latest.mobile.#']} using route.index.gecko.v2.mozilla-aurora.nightly.latest.mobile.#
2016-10-20 15:12:26,615 - channel.py - DEBUG - Queue bound
2016-10-20 15:12:26,820 - worker.py - INFO - Worker has completed running.
2016-10-20 15:12:26,841 - base_events.py - ERROR - Task was destroyed but it is pending!
task: <Task pending coro=<AmqpProtocol.run() running at /home/jlorenzo/.virtualenvs/nightlies-watcher/lib/python3.5/site-packages/aioamqp/protocol.py:284> wait_for=<Future pending cb=[Task._wakeup()]>>
```